invenio_communities/members/services/service.py

```diff
@@ -56,14 +65,148 @@ def read(self, identity, id_):
+        data, _ = self.schema.load(
+            data,
+            context=dict(
+                identity=identity,
+                record=member_record
+            )
+        )
+
+        self._assert_can_change_role(identity, member_record)
+
+        member_record.update(
+            dict(
+                community_id=data.get("community"),
+                user_id=data.get("user"),
+                role=data.get("role")
+            )
+        )
+
+        uow.register(RecordCommitOp(member_record, self.indexer))
+
+        return self.result_item(
+            self,
+            identity,
+            member_record,
+            links_tpl=self.links_item_tpl,
+        )
+
+    def _assert_can_change_role(self, identity, member_record):
+        """Raises if identity can't change role."""
+        # Owner/Manager trying to change their role
+        # TODO? account for group maybe (we do want person owner though...)
+        if identity.id == member_record.user_id:
+            if member_record.role == "owner":
+                raise OwnerSelfRoleChangeError()
+            if member_record.role == "manager":
+                raise ManagerSelfRoleChangeError()
+
+        # Manager trying to change owner
+        manager_need = create_community_role_need(
+            member_record.community_id, "manager"
+        )
+        if manager_need in identity.provides and member_record.role == "owner":
+            raise PermissionDeniedError()
+
+    @unit_of_work()
+    def delete(self, identity, id_, revision_id=None, uow=None):
+        """Delete member."""
+        member_record = self.record_cls.get_record(id_)
+
+        self.check_revision_id(member_record, revision_id)
+
+        # Permissions
+        self.require_permission(
+            identity, "delete_member", record=member_record
+        )
+
+        # Check if last owner
+        if self._deleting_last_owner(identity, member_record):
+            raise LastOwnerError()
+
+        # Run components
+        self.run_components(
+            'delete_member', identity, record=member_record, uow=uow
+        )
+
+        uow.register(
+            RecordDeleteOp(member_record, self.indexer, index_refresh=True)
+        )
+
+        return True
+
+    def _deleting_last_owner(self, identity, member_record):
+        """Check if member_record is last owner of its community."""
+        if member_record.role != "owner":
+            return False
+
+        search = self._search(
+            'search_members',
+            identity,
+            params={},
+            es_preference=None,
+            permission_action='read_search_members',
+            extra_filter=(
+                Q('term', community_id=str(member_record.community_id)) &
+                Q('term', role="owner")
+            )
+        )
+
+        owners = [h for h in search.execute()]
+
+        return len(owners) == 1
+
+    def search(
+            self, identity, community_id, params=None, es_preference=None,
```

Comment:
To note: I am leaning towards having this method build the extra_filter needed based on the passed community_id (/uuid in future). 

Also no community_id seems like a possibility too if we programmatically want to search on ALL memberships. But maybe `_search()` is the way to go then...